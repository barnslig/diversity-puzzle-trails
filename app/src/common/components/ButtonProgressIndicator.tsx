import { CircularProgress } from "@mui/material";
import * as React from "react";

/**
 * Circular progress that can be put into a <Button /> while loading
 */
const ButtonProgressIndicator = () => (
  <CircularProgress
    size={24}
    sx={{
      position: "absolute",
      top: "50%",
      left: "50%",
      marginTop: "-12px",
      marginLeft: "-12px",
    }}
  />
);

export default ButtonProgressIndicator;
