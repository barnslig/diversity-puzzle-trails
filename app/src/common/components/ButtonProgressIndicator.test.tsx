import * as React from "react";
import render from "../testing/render";

import ButtonProgressIndicator from "./ButtonProgressIndicator";

it("renders correctly", () => {
  const { container } = render(<ButtonProgressIndicator />);
  expect(container.firstChild).toMatchSnapshot();
});
