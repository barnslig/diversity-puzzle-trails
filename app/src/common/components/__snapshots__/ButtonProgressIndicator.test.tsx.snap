// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1g9nplg-MuiCircularProgress-root"
  role="progressbar"
  style="width: 24px; height: 24px;"
>
  <svg
    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
      cx="44"
      cy="44"
      fill="none"
      r="20.2"
      stroke-width="3.6"
    />
  </svg>
</span>
`;
